src/main/java/seedu/weme/logic/prompter/MemeAddCommandPrompter.java

```diff
@@ -0,0 +1,33 @@
+package seedu.weme.logic.prompter;
+
+import static seedu.weme.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.weme.logic.parser.ArgumentTokenizer.getLastArgument;
+import static seedu.weme.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.weme.logic.parser.CliSyntax.PREFIX_FILEPATH;
+import static seedu.weme.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.weme.logic.prompter.PrompterUtil.findSimilarArguments;
+
+import seedu.weme.logic.commands.MemeAddCommand;
+import seedu.weme.logic.prompter.exceptions.PromptException;
+import seedu.weme.model.Model;
+
+/**
+ * Prompt arguments for MemeAddCommand.
+ */
+public class MemeAddCommandPrompter implements Prompter {
+    @Override
+    public CommandPrompt prompt(Model model, String arguments) throws PromptException {
+        if (arguments.isBlank()) {
+            return CommandPrompt.empty();
+        }
+
+        LastArgument lastArgument = getLastArgument(arguments, PREFIX_FILEPATH, PREFIX_DESCRIPTION, PREFIX_TAG);
+        if (arguments.length() > 2 && lastArgument == null) {
```

Comment:
2 can be abstracted out here, and maybe use `requireNonNull` for`lastArgument`.

Reply:
`lastArgument` can be null, which indicates that no prefix is found in the input.